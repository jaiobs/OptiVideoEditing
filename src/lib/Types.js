export const ORIENTATION = 'orientation';
export const PICTURE_TAKEN = 'picture_taken';
export const VIDEO_TAKEN = 'video_taken';
export const BACK_PRESS_VIDEO = 'back_press_video';

export const STICKER_CATEGORIES_SUCCESS = 'Sticker_categories_Success';
export const STICKER_CATEGORIES_FAILURE = 'Sticker_categories_Failure';
export const STICKER_LIST_SUCCESS = 'Sticker_list_Success';
export const STICKER_LIST_FAILURE = 'Sticker_list_Failure';
export const FLASH_ACTION = 'flash_action';
export const SWITCH_CAMERA_ACTION = 'switch_camera_action';

// STICKER URL

//----Fetch all sticker categories-----//
export const STICKER_CATOGORIES = 'v0/post/sticker-category-listing';

//----Fetch all stickers list based on categories------//
export const FETCH_ALL_STICKERS = 'v0/post/sticker-listing?id=';

//MUSIC SEARCH
export const TRACK_LIST_FETCH = 'track_list_fetch';

export const SOUND_CLOUD_BASE_URL = 'https://api.soundcloud.com';
export const GET_TRACKS = '/tracks';

// TAGGING
export const TAG_ALL_USERS_LISTING = 'v0/user/listing-n-search';
export const TAG_FRIENDS_LISTING = 'v0/user/friends-listing';
export const TAG_ALL_USERS_FETCH = 'v0/tag_all_users_fetch';

//RETAIN VIDEO IN PREVIEW PAGE
export const RETAIN_VIDEO_SEGMENTS = 'retain_video_segments';

//OVERLAY DATA
export const OVERLAY_DATA_LIST = 'overlay_data_list';
