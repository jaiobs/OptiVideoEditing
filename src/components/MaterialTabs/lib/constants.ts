export default {
  barHeight: 48,
  indicatorHeight: 4,
};
