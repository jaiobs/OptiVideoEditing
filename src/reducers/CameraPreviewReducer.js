import {
  ORIENTATION,
  PICTURE_TAKEN,
  VIDEO_TAKEN,
  RETAIN_VIDEO_SEGMENTS,
  BACK_PRESS_VIDEO,
  FLASH_ACTION,
  SWITCH_CAMERA_ACTION,
} from '../lib/Types';

const INITIAL_STATE = {
  orientationCheck: 'portrait',
  onPictureTaken: {},
  onVideoTaken: {},
  retainVideoSegments: [],
  backPressVideoDetails: {},
  flashDetails: false,
  switchCameraDetails: false,
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case ORIENTATION:
      return {
        ...state,
        orientationCheck: action.payload,
      };

    case PICTURE_TAKEN:
      return {
        ...state,
        onPictureTaken: action.payload,
      };

    case VIDEO_TAKEN:
      return {
        ...state,
        onVideoTaken: action.payload,
      };

    case BACK_PRESS_VIDEO:
      return {
        ...state,
        backPressVideoDetails: action.payload,
      };

    case RETAIN_VIDEO_SEGMENTS:
      return {
        ...state,
        retainVideoSegments: action.payload,
      };

    case FLASH_ACTION:
      return {
        ...state,
        flashDetails: action.payload,
      };

    case SWITCH_CAMERA_ACTION:
      return {
        ...state,
        switchCameraDetails: action.payload,
      };

    default:
      return state;
  }
};
