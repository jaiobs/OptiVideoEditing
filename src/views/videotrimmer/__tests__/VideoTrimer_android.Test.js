import React from 'react';
import renderer from 'react-test-renderer';
import VideoTrimer from '../VideoTrimer.ios';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

import CameraPreviewReducer from '../../../reducers/CameraPreviewReducer';
import * as types from '../../../lib/Types';

jest.useFakeTimers();

const mockOrientation = 'portrait';
const mockStore = configureMockStore({});

describe('Android orientationCheck reducer', () => {
  it('should return the initial state', () => {
    expect(CameraPreviewReducer(undefined, {})).toEqual({
      orientationCheck: mockOrientation,
      backPressVideoDetails: {},
      flashDetails: false,
      onPictureTaken: {},
      onVideoTaken: {},
      retainVideoSegments: [],
      switchCameraDetails: false,
    });
  });

  it('should handle android orientation action payload', () => {
    expect(
      CameraPreviewReducer('', {
        type: types.ORIENTATION,
        payload: mockOrientation,
      }),
    ).toEqual({
      orientationCheck: mockOrientation,
    });
  });
});

describe('android VideoTrimerTest', () => {
  it('should render correctly with state to props', () => {
    const initialState = {
      CameraPreviewReducer: {orientationCheck: mockOrientation},
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <VideoTrimer navigation={{getParam: jest.fn()}} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
