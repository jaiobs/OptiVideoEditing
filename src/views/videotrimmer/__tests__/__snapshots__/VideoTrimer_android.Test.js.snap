// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`android VideoTrimerTest should render correctly with state to props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "position": "absolute",
        },
        undefined,
      ]
    }
  />
  <VideoTrimmer
    filter={
      Object {
        "brightness": 0,
        "contrast": 1,
        "name": "",
        "range": 1,
        "saturation": 1,
        "type": "NORMAL",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 1034,
        "position": "absolute",
        "right": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "maxHeight": 50,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "left": 15,
              "opacity": 1,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "position": "absolute",
              "top": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/res/assets/images/cross.png",
              }
            }
            style={
              Object {
                "height": 24,
                "resizeMode": "contain",
                "width": 24,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "textShadowColor": "rgba(0, 0, 0, 0.75)",
              "textShadowOffset": Object {
                "height": 1,
                "width": -1,
              },
              "textShadowRadius": 10,
            }
          }
        >
          tap screen for filters
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textShadowColor": "rgba(0, 0, 0, 0.75)",
              "textShadowOffset": Object {
                "height": 1,
                "width": -1,
              },
              "textShadowRadius": 10,
            }
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "bottom": 180,
          "flexDirection": "row",
          "height": "100%",
          "marginTop": 50,
          "position": "absolute",
          "width": "80%",
          "zIndex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 18,
        "height": 50,
        "justifyContent": "center",
        "position": "absolute",
        "right": 0,
        "width": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "flex-end",
          "height": 32,
          "margin": 10,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 13,
          "opacity": 1,
          "width": 32,
        }
      }
      testID="nextBtn_on_Trimmer"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/res/assets/images/next_button_send.png",
          }
        }
        style={
          Object {
            "height": 32,
            "resizeMode": "contain",
            "width": 32,
          }
        }
      />
    </View>
  </View>
</View>
`;
